tests/test_sync.py

```diff
@@ -257,8 +263,89 @@ def test_get_sync_work_state(self):
                 with self.subTest(part_size=part_size, object_size=event['source_obj_metadata']['size']):
                     self.assertEqual(sync.get_sync_work_state(event)['total_parts'], expected_total_parts)
 
+    def generate_random_blob_key(self):
+        random_uuid = str(uuid.uuid4())
+        sha1 = hashlib.sha1(str.encode(random_uuid)).hexdigest()
```

Comment:
Please randomly include the S3 extension for multipart objects.

Reply:
s3 multipart uploads use the same key, but change the query paramters on the URL to include the UploadID + part number. https://github.com/HumanCellAtlas/data-store/blob/69d412cfc1ecd2c83dfdb597b276f024dcbee3a7/daemons/dss-sync-sfn/app.py#L291

Reply:
blob keys can have a `-{number_of_parts}` within their `aws_etag` section, added a test to ensure that they don't get filtered out.